test/terraform_aws_lambda_example_test.go

```diff
@@ -62,6 +62,83 @@ func TestTerraformAwsLambdaExample(t *testing.T) {
 	assert.Contains(t, string(functionError.Payload), "Failed to handle")
 }
 
+// Annother example of how to test the Terraform module in
+// examples/terraform-aws-lambda-example using Terratest, this time with
+// the aws.InvokeFunctionWithParams.
+func TestTerraformAwsLambdaWithParamsExample(t *testing.T) {
+	t.Parallel()
+
+	// Give this lambda function a unique ID for a name so we can distinguish it from any other lambdas
+	// in your AWS account
+	functionName := fmt.Sprintf("terratest-aws-lambda-withparams-example-%s", random.UniqueId())
+
+	// Pick a random AWS region to test in. This helps ensure your code works in all regions.
+	awsRegion := aws.GetRandomStableRegion(t, nil, nil)
+
+	// Construct the terraform options with default retryable errors to handle the most common retryable errors in
+	// terraform testing.
+	terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+		// The path to where our Terraform code is located
+		TerraformDir: "../examples/terraform-aws-lambda-example",
```

Comment:
Since these tests run in parallel, if you run more than one test in a given folder, it's possible for the state files and `.terraform` folders to clobber each other. Therefore, you'll need to use [this function](https://github.com/gruntwork-io/terratest/blob/master/modules/test-structure/test_structure.go#L70) in this test and the one above it to make a copy of this folder.

Reply:
Done